### PR TITLE
added ReactionController with create and find endpoints

### PR DESCRIPTION
**Pull request by vorobevamv**

### CODE REVIEW

File: src/ImageService.Business/Commands/ImageNews/CreateImageNewsCommand.cs [hunk 1]
```diff
@@ -54,7 +51,6 @@ public async Task<OperationResultResponse<CreateImageNewsResponse>> ExecuteAsync
 
         return new OperationResultResponse<CreateImageNewsResponse>
         {
-          Status = OperationResultStatusType.Failed,
           Errors = new() { "Not enough rights." }
```
Comment (MariyaSin) [inline src/ImageService.Business/Commands/ImageNews/CreateImageNewsCommand.cs:54]: please? use response creator

File: src/ImageService.Business/Commands/ImageNews/CreateImageNewsCommand.cs [hunk 2]
```diff
@@ -65,7 +61,6 @@ public async Task<OperationResultResponse<CreateImageNewsResponse>> ExecuteAsync
 
         return new OperationResultResponse<CreateImageNewsResponse>
```
Comment (MariyaSin) [inline src/ImageService.Business/Commands/ImageNews/CreateImageNewsCommand.cs:62]: same

File: src/ImageService.Business/Commands/ImageReaction/FIndImageReactionCommand.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      if (!validationResult.IsValid)
+      {
+        return _responseCreator.CreateFailureFindResponse<ReactionInfo>(HttpStatusCode.BadRequest,
+          validationResult.Errors.Select(vf => vf.ErrorMessage).ToList());
+      }
+
+      FindResultResponse<ReactionInfo> response = new();
+      (List<DbImage> dbRectionList, int totalCount) = await _repository.FindReactionAsync(findReactionFilter);
+
+      response.Body = dbRectionList?.Select(dbImage => _mapper.Map(dbImage)).ToList();
```
Comment (MariyaSin) [inline src/ImageService.Business/Commands/ImageReaction/FIndImageReactionCommand.cs:49]: use ..(_mapper.Map)..

File: src/ImageService.Data/ImageRepository.cs [hunk 4]
```diff
@@ -24,10 +25,37 @@ private string GetTargetDBTableName(ImageSource imageSource)
...
         case ImageSource.News: return DBTablesNames.NEWS;
         case ImageSource.Education: return DBTablesNames.EDUCATION;
+        case ImageSource.Reaction: return DBTablesNames.REACTION;
         default: throw new ArgumentOutOfRangeException();
       }
     }
 
+    private IQueryable<DbImage> CreateFindReactionPredicates(
+      FindReactionFilter filter,
+      IQueryable<DbImage> dbReactionList)
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageRepository.cs:35]: rename to query

File: src/ImageService.Data/ImageRepository.cs [hunk 5]
```diff
@@ -89,5 +117,29 @@ SELECT ParentId
...
+      if (filter is null)
+      {
+        return (null, default);
+      }
+
+      IQueryable<DbImage> dbReactionList = CreateFindReactionPredicates(
+        filter,
+        _provider
+        .FromSqlRaw($"SELECT * FROM {GetTargetDBTableName(ImageSource.Reaction)}")
+            .OrderByDescending(x => x.CreatedAtUtc)
```
Comment (VLiamin) [inline src/ImageService.Data/ImageRepository.cs:132]: spaces

File: src/ImageService.Data/Interfaces/IImageRepository.cs [hunk 6]
```diff
@@ -17,5 +18,8 @@ public interface IImageRepository
     Task<DbImage> GetAsync(ImageSource sourse, Guid imageId);
 
     Task RemoveAsync(ImageSource sourse, List<Guid> imagesIds);
+
+    Task<(List<DbImage> dbReactions, int totalCount)> FindReactionAsync(FindReactionFilter filter);
+    Task<bool> DoesSameNameExistAsync(string name, ImageSource source);
```
Comment (VLiamin) [inline src/ImageService.Data/Interfaces/IImageRepository.cs:23]: make empty line

File: src/ImageService.Business/Commands/ImageNews/CreateImageNewsCommand.cs (hunk 1 above)
Comment (vorobevamv) [reply src/ImageService.Business/Commands/ImageNews/CreateImageNewsCommand.cs:54]: done

File: src/ImageService.Business/Commands/ImageNews/CreateImageNewsCommand.cs (hunk 2 above)
Comment (vorobevamv) [reply src/ImageService.Business/Commands/ImageNews/CreateImageNewsCommand.cs:62]: done

File: src/ImageService.Business/Commands/ImageReaction/FIndImageReactionCommand.cs (hunk 3 above)
Comment (vorobevamv) [reply src/ImageService.Business/Commands/ImageReaction/FIndImageReactionCommand.cs:49]: I'm not sure if understood your comment right..
Please, check

File: src/ImageService.Data/ImageRepository.cs (hunk 4 above)
Comment (vorobevamv) [reply src/ImageService.Data/ImageRepository.cs:35]: done

File: src/ImageService.Data/ImageRepository.cs (hunk 5 above)
Comment (vorobevamv) [reply src/ImageService.Data/ImageRepository.cs:132]: fixed

File: src/ImageService.Data/Interfaces/IImageRepository.cs (hunk 6 above)
Comment (vorobevamv) [reply src/ImageService.Data/Interfaces/IImageRepository.cs:23]: fixed

**Pull request closed without merging**